lib/driver.js

```diff
@@ -1090,7 +1215,100 @@ export class XCUITestDriver extends BaseDriver {
     if (cmd === 'getStatus') {
       return await this.getStatus();
     }
-    return await super.executeCommand(cmd, ...args);
+
+    if (cmd === 'createSession') {
+      return await super.executeCommand(cmd, ...args);
+    }
+
+    const wdaRestartEnabled = Boolean(this.opts.enableWdaRestart);
+
+    if (!wdaRestartEnabled) {
+      return await super.executeCommand(cmd, ...args);
+    }
+
+    const maxAttempts = this.opts.wdaStartupRetries || 2;
+    let attempts = 0;
+
+    while (true) {
```

Comment:
```suggestion
    while (attempts < maxAttempts) {
```
(no much meanings, but just to avoid `while (true)`)